packages/mask/src/extension/popups/hooks/useFriendsFromSearch.tsx

```diff
@@ -1,43 +1,82 @@
-import { ECKeyIdentifier, EMPTY_LIST, type NextIDPersonaBindings } from '@masknet/shared-base'
+import {
+    ECKeyIdentifier,
+    EMPTY_LIST,
+    type NextIDPersonaBindings,
+    type EnhanceableSite,
+    type ProfileIdentifier,
+} from '@masknet/shared-base'
 import { uniqBy } from 'lodash-es'
 import { useMemo } from 'react'
 import type { Friend } from './useFriends.js'
 import { useCurrentLinkedPersona } from '@masknet/shared'
 import { profilesFilter } from './useFriendProfiles.js'
-import { PlatformSort } from '../pages/Friends/common.js'
+import { PlatformSort, type Profile } from '../pages/Friends/common.js'
 
-export type NextIDPersonaBindingsWithIdentifier = NextIDPersonaBindings & { linkedPersona: ECKeyIdentifier } & {
+export type NextIDPersonaBindingsWithIdentifier = Omit<NextIDPersonaBindings, 'proofs'> & { proofs: Profile[] } & {
+    linkedPersona: ECKeyIdentifier
+} & {
     isLocal?: boolean
 }
 
 export function useFriendsFromSearch(
+    localSearchedResult?: Friend[],
```

Comment:
I see you pass a NonNullable array to this hook. Let TypeScript help you to guard it.

```suggestion
    localSearchedResult: Friend[],
```